KakapoExample/KakapoTests/KakapoServerTests.swift

```diff
@@ -107,6 +107,83 @@ class KakapoServerTests: QuickSpec {
             }
+            
+            it("should not call the handler when requesting a registered url but passing different HTTPMethod") {
+                var info: URLInfo? = nil
+                var responseURL: NSURL? = NSURL(string: "")
+                var responseError: NSError? = NSError(domain: "", code: 1, userInfo: nil)
+                
+                KakapoServer.del("/users/:id") { request in
+                    // Shouldn't reach here
+                    info = request.info
+                }
+                
+                let request = NSMutableURLRequest(URL: NSURL(string: "/users/1")!)
+                request.HTTPMethod = "PUT"
+                NSURLSession.sharedSession().dataTaskWithRequest(request) { (_, response, error) in
+                    // Response will be nil since error
+                    responseURL = response?.URL
+                    responseError = error
+                    }.resume()
+                
+                expect(info?.params).toEventually(beNil())
+                expect(info?.queryParams).toEventually(beNil())
+                expect(responseURL?.absoluteString).toEventually(beNil())
+                expect(responseError).toNotEventually(beNil())
+            }
+            
+            it("should give back the body in the handler when the request is setting it - NSURLSession") {
+                var info: URLInfo? = nil
+                var bodyData: NSData? = nil
+                var bodyDictionary: NSDictionary? = nil
+                
+                let request = NSMutableURLRequest(URL: NSURL(string: "/user_equipment/1")!)
+                request.HTTPMethod = "POST"
+                let params = ["username":"test", "password":"pass"] as Dictionary<String, String>
+                request.HTTPBody = try? NSJSONSerialization.dataWithJSONObject(params, options: .PrettyPrinted)
+                request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+                request.addValue("application/json", forHTTPHeaderField: "Accept")
+                
+                KakapoServer.post("/user_equipment/:id") { request in
+                    info = request.info
+                    bodyData = request.HTTPBody
+                    bodyDictionary = try! NSJSONSerialization.JSONObjectWithData(bodyData!, options: .MutableLeaves) as? NSDictionary
+                }
+                
+                
+                NSURLSession.sharedSession().dataTaskWithRequest(request){ (_, _, _) in }.resume()
+                
+                expect(info?.params).toEventually(equal(["id" : "1"]))
+                expect(info?.queryParams).toEventually(equal([ : ]))
+                expect(bodyData).toNotEventually(beNil())
+                expect(bodyDictionary).toNotEventually(beNil())
+            }
+            
+            it("should give back the body in the handler when the request is setting it - NSURLConnection") {
```

Comment:
could you name this and the previous differently? 
e.g.
"it should give back the body in the handler when a NSURLConnection has it..."
"it should give back the body in the handler when a NSURLSession has it..."

that "- Classname" is weird and not as easy to understand